modules/codegen/src/test/scala/tests/generators/akkaHttp/AkkaHttpServerTest.scala

```diff
@@ -397,4 +397,140 @@ class AkkaHttpServerTest extends AnyFunSuite with Matchers with SwaggerSpecRunne
     genHandler.structure should equal(handler.structure)
     genResource.structure should equal(resource.structure)
   }
+
+  test("Ensure routes are generated with custom extraction") {
+    val (
+      _,
+      _,
+      Servers(Server(pkg, extraImports, genHandler, genResource :: Nil) :: Nil, Nil)
+    ) = runSwaggerSpec(swagger)(Context.empty.copy(customExtraction = true), AkkaHttp)
+
+    val handler  = q"""
+      trait StoreHandler[E] {
+        def getRoot(respond: StoreResource.GetRootResponse.type)()(extracted: E): scala.concurrent.Future[StoreResource.GetRootResponse]
+        def putBar(respond: StoreResource.PutBarResponse.type)(bar: Long)(extracted: E): scala.concurrent.Future[HttpResponse]
+        def getFoo(respond: StoreResource.GetFooResponse.type)()(extracted: E): scala.concurrent.Future[StoreResource.GetFooResponse]
+        def getFooBar(respond: StoreResource.GetFooBarResponse.type)(bar: Long)(extracted: E): scala.concurrent.Future[StoreResource.GetFooBarResponse]
+        def getOrderById(respond: StoreResource.GetOrderByIdResponse.type)(orderId: Long, status: OrderStatus = OrderStatus.Placed)(extracted: E): scala.concurrent.Future[StoreResource.GetOrderByIdResponse]
+      }
+    """
+    val resource = q"""
+      object StoreResource {
+        def routes[E](handler: StoreHandler[E], customExtract: String => Directive1[E])(implicit mat: akka.stream.Materializer): Route = {
```

Comment:
The TL;DR for this PR: enabled `customExtract` here, and `extracted` in the above handler trait.